experimental/framework/frs-client/src/interfaces.ts

```diff
@@ -0,0 +1,57 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IMember, IServiceAudience } from "@fluid-experimental/fluid-static";
+import { ITelemetryBaseLogger } from "@fluidframework/common-definitions";
+import { ITokenProvider } from "@fluidframework/routerlicious-driver";
+
+export interface FrsContainerConfig {
+    id: string;
+    logger?: ITelemetryBaseLogger;
+}
+
+interface FrsConnectionConfigBase {
+    tenantId: string;
+    orderer: string;
+    storage: string;
+    user?: FrsMember;
+}
+
+interface FrsConnectionConfigWithKey extends FrsConnectionConfigBase {
+    key: string;
+    tokenProvider?: never;
```

Comment:
Do you actually need these `never`s?  Looks like you should just be able to drop the properties.

Reply:
I needed the `never`s here to make it an explicit either/or between the key and the tokenProvider. If we just drop the properties, then ts lint will complain about the other value not being found when we try to access it.
For example, trying to access `config.tokenProvider` gives `Property 'tokenProvider' does not exist on type 'FrsConnectionConfigWithKey'.`

Reply:
If you put a `type: "key"` and `type: "tokenProvider"` on the respective connection config types, can you use discriminated union here instead?  With the `never` approach all types must know about all properties of all other types.

Reply:
Updated to use discriminated union!